### PR TITLE
Gradle 6.8 upgrade

### PR DESCRIPTION
**Pull request by vRallev**

### CODE REVIEW

**Comment by vRallev** [comment]
I guess we need to wait until Gradle migrated to 1.4 itself.

File: gradle-plugin/build.gradle [hunk 1]
```diff
@@ -70,18 +70,16 @@ tasks.withType(JavaCompile).configureEach {
...
   }
 }
 
 dependencies {
   implementation deps.kotlin.gradle_plugin
   implementation deps.kotlin.gradle_plugin_api
-  implementation deps.kotlin.stdlib
+
+  // Necessary to bump a transitive dependency.
+  implementation deps.kotlin.reflect
```
Comment (vRallev) [inline gradle-plugin/build.gradle:82]: Should probably be `compileOnly`.

File: gradle-plugin/build.gradle [hunk 2]
```diff
@@ -70,18 +70,16 @@ tasks.withType(JavaCompile).configureEach {
...
+    allWarningsAsErrors = true
   }
 }
 
 dependencies {
   implementation deps.kotlin.gradle_plugin
   implementation deps.kotlin.gradle_plugin_api
-  implementation deps.kotlin.stdlib
+
+  // Necessary to bump a transitive dependency.
```
Comment (christiandeange) [inline gradle-plugin/build.gradle:81]: Nice lol

**Review by christiandeange — APPROVED** [review]

**Comment by vRallev** [comment]
This PR is open since August last year... Gradle was fast to upgrade Kotlin </rant>

**Review by christiandeange — APPROVED** [review]

**Pull request merged**